### PR TITLE
Updating .gitignore file

### PR DESCRIPTION
**Pull request by arcanedev-maroc**

### CODE REVIEW

**Comment by maiorano84** [comment]
I'm of the belief that composer.lock should not be added to the .gitignore file. This is what ensures everybody is using the same version of all dependencies that are installed.

**Comment by arcanedev-maroc** [comment]
@maiorano84 the actual `composer.lock` is outdated and does not match the dependencies mentioned in `composer.json` (like `illuminate/support`).

And it is recommended for your own apps, not for frameworks/packages.

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,5 @@
-.DS_Store
-.idea
\ No newline at end of file
+/.idea/
+/vendor/
+/.DS_Store
```
Comment (arcanedev-maroc) [inline .gitignore:3]: I think `.DS_Store` must be included in global gitignore instead

**Comment by tnylea** [comment]
Hey @arcanedev-maroc Thanks for the contributions. I agree that the composer.lock should not be included in the gitignore. Go ahead and submit another PR and I can merge that :)

Thanks.

**Comment by arcanedev-maroc** [comment]
@tnylea Just reopen this one, i can edit this PR.

**Pull request closed without merging**